platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/DefaultConfigurationResolver.java

```diff
@@ -219,7 +219,7 @@ public ResolverResults resolveBuildDependencies(ResolveContext resolveContext) {
             // set of build dependencies than actually required. This is because it takes a lot of extra information
             // from the visited graph to properly filter artifacts by dependencySpec, and we don't want capture that when
             // calculating build dependencies.
-            dependencySpec -> visitedArtifacts.select(getImplicitSelectionSpec(resolveContext))
+            dependencySpec -> visitedArtifacts.select(getImplicitSelectionSpec(resolveContext), false)
```

Comment:
❓ What to do here?  I assume we want the old behavior, which is to treat this as not lenient.